src/main/scala/org/globalforestwatch/features/FeatureRDD.scala

```diff
@@ -41,19 +53,53 @@ object FeatureRDD {
     val scalaRDD =
       org.apache.spark.api.java.JavaRDD.toRDD(spatialRDD.spatialPartitionedRDD)
 
-    scalaRDD.map {
-      case pt: GeoSparkPoint =>
-        val pointFeatureData = pt.getUserData.asInstanceOf[String].split('\t')
+    scalaRDD
+      .map {
+        case pt: GeoSparkPoint =>
+          val pointFeatureData = pt.getUserData.asInstanceOf[String].split('\t')
+
+          val geom = GeometryReducer.reduce(GeometryReducer.gpr)(
+            vector.Point(pt.getX, pt.getY)
+          )
+
+          val pointFeatureId: FeatureId =
+            featureObj.getFeatureId(pointFeatureData)
+          vector.Feature(geom, pointFeatureId)
+        case _ =>
+          throw new NotImplementedError(
+            "Cannot convert geometry type to Geotrellis RDD"
+          )
+      }
+    // In case we implement this method for other geometry types we will have to split geometries
+    //      .flatMap { feature =>
+    //        splitGeometry(feature)
+    //      }
+
+  }
+
+  private def splitGeometry(
```

Comment:
Instead of doing all this intersecting sequentially for each tile, why don't you just use read all the features and grid tiles into two GeoSpark RDDs and do a spatial join?